src/views/ops/Job/components/BaseJob.vue

```diff
@@ -0,0 +1,167 @@
+            }
+          },
+          comment: {
+            width: '240px'
+          },
+          summary: {
+            label: this.$t('Summary'),
+            formatter: (row) => {
+              return row.summary['success'] + '/' + row.summary['total']
+            }
+          },
+          average_time_cost: {
+            formatter: (row) => {
+              return row.average_time_cost.toFixed(2) + 's'
+            }
+          },
+          asset_amount: {
+            label: this.$t('AssetsOfNumber'),
+            formatter: (row) => {
+              return row.assets.length
+            }
+          },
+          date_last_run: {
+            width: '140px',
+            formatter: DateFormatter
+          },
+          actions: {
+            formatter: ActionsFormatter,
+            formatterArgs: {
+              hasUpdate: true,
+              canUpdate: this.$hasPerm('ops.change_job') && !this.$store.getters.currentOrgIsRoot,
+              updateRoute: 'JobUpdate',
+              hasDelete: true,
+              canDelete: this.$hasPerm('ops.delete_job'),
+              hasClone: false,
+              extraActions: [
+                {
+                  title: this.$t('Run'),
+                  name: 'run',
+                  order: 5,
+                  type: 'primary',
+                  can: this.$hasPerm('ops.add_jobexecution') && !this.$store.getters.currentOrgIsRoot,
+                  callback: ({ row }) => {
+                    this.item = row
+                    if (row?.use_parameter_define && row?.parameters_define) {
+                      const params = JSON.parse(row.parameters_define)
+                      if (Object.keys(params).length > 0) {
+                        this.showJobRunDialog = true
+                      }
+                    } else if (row?.variable?.length) {
+                      this.showVariableDialog = true
+                    } else {
+                      this.runJob(row)
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        }
+      },
+      headerActions: {
+        createRoute: () => {
+          return {
+            name: 'JobCreate',
+            query: {
+              type: this.type
+            }
+          }
+        },
+        hasRefresh: true,
+        hasExport: false,
+        hasImport: false
+      },
+      showJobRunDialog: false,
+      showVariableDialog: false
+    }
+  },
+  computed: {
+    formData() {
+      return this.item.variable.map((data) => {
+        return data.form_data
+      })
+    }
+
+  },
+  methods: {
+    runJob(row, parameters) {
+      this.$axios.post('/api/v1/ops/job-executions/', {
+        job: row.id,
+        parameters: parameters
+      }).then((resp) => {
+        this.showVariableDialog = false
+        openTaskPage(resp.task_id)
+      })
+    },
+    runJobWithParams(parameters) {
+      this.runJob(this.item, parameters)
+    }
+  }
+}
+</script>
```

Comment:
No issues or regularities found.

In terms of potential optimizations, consider optimizing repetitive logic such as loading items from API to reduce HTTP requests or using memoization if it applies.